puppet/modules/redmine/manifests/init.pp

```diff
@@ -86,44 +133,64 @@
     servername              => $servername,
   }
 
+  if $https {
+    apache::vhost { $servername:
+      add_default_charset     => 'UTF-8',
+      docroot                 => $docroot,
+      manage_docroot          => false,
+      port                    => 443,
+      options                 => ['SymLinksIfOwnerMatch'],
+      passenger_app_root      => $app_root,
+      passenger_min_instances => $min_instances,
+      passenger_start_timeout => $start_timeout,
+      priority                => $priority,
+      servername              => $servername,
+      ssl                     => true,
+      ssl_cert                => "/etc/letsencrypt/live/${servername}/fullchain.pem",
+      ssl_chain               => "/etc/letsencrypt/live/${servername}/chain.pem",
+      ssl_key                 => "/etc/letsencrypt/live/${servername}/privkey.pem",
+      require                 => Letsencrypt::Certonly[$servername],
+    }
+  }
+
   file { ["${app_root}/config.ru", "${app_root}/config/environment.rb"]:
-    owner => $db_username,
+    owner => $username,
   }
 
   # cron jobs ported from .openshift
 
   file { '/etc/cron.daily/redmine_backup':
-    ensure  => present,
+    ensure  => file,
     owner   => 'root',
     group   => 'root',
     mode    => '0755',
-    content => template('redmine/postgresql_backup.sh')
+    content => template('redmine/postgresql_backup.sh'),
   }
 
   file { '/usr/local/bin/redmine_repos.sh':
-    ensure  => present,
+    ensure  => file,
     owner   => 'root',
     group   => 'root',
     mode    => '0755',
-    content => template('redmine/git_repos.sh')
+    content => file('redmine/git_repos.sh'),
   }
 
   file { '/etc/cron.hourly/redmine_repos':
-    ensure  => present,
+    ensure  => file,
     owner   => 'root',
     group   => 'root',
     mode    => '0755',
     content => "#!/bin/bash
-sudo -u ${db_username} /usr/local/bin/redmine_repos.sh"
+sudo -u ${username} /usr/local/bin/redmine_repos.sh ${app_root} ${data_dir}",
```

Comment:
Good call, this will make it easy to return the cron scripts to the redmine git repo if we want...